Implement a C.I build

Maybe using AppVeyor, i.e. like this https://ci.appveyor.com/project/mattwarren/minibench-warren. This way if someone (probably me) breaks the build or some tests, we'll know about it sooner rather than later

See [this discussion](https://gitter.im/PerfDotNet/BenchmarkDotNet?at=569d478b5de13b3f15e3c9d0) for some background and extra info.

Thanks to both of you for the tip about `--fallbacksource https://dotnet.myget.org/F/dotnet-core/api/v3/index.json` that worked (takes a [long time to restore though](https://ci.appveyor.com/project/mattwarren/benchmarkdotnet-8ww0x/build/1.0.6#L857) 239423ms! I hope AppVeyor does some caching and isn't downloading all that for every build!!

However I now have a problem related to MsBuild, seems like MsBuild 12.0 isn't installed:

```
Compiling BenchmarkDotNet for .NETFramework,Version=v4.5 
C:\Program Files\dotnet\bin\dotnet.exe compile-csc @C:\projects\benchmarkdotnet-8ww0x\BenchmarkDotNet\obj\Debug\net45\dotnet-compile.rsp returned Exit Code 1 
C:\projects\benchmarkdotnet-8ww0x\BenchmarkDotNet\project.json(35,28): warning NU1007: Dependency specified was Microsoft.Build >= 12.0.0 but ended up with Microsoft.Build 4.0.0.0. 
C:\projects\benchmarkdotnet-8ww0x\BenchmarkDotNet\project.json(36,38): warning NU1007: Dependency specified was Microsoft.Build.Framework >= 12.0.0 but ended up with Microsoft.Build.Framework 4.0.0.0. 
C:\projects\benchmarkdotnet-8ww0x\BenchmarkDotNet\project.json(37,44): error NU1001: The dependency Microsoft.Build.Utilities.v12.0 >= 12.0.0 could not be resolved. 
C:\projects\benchmarkdotnet-8ww0x\BenchmarkDotNet\BenchmarkDotNet\Loggers\MSBuildConsoleLogger.cs(3,23): error CS0234: The type or namespace name 'Utilities' does not exist in the namespace 'Microsoft.Build' (are you missing an assembly reference?) 
C:\projects\benchmarkdotnet-8ww0x\BenchmarkDotNet\BenchmarkDotNet\Loggers\MSBuildConsoleLogger.cs(7,43): error CS0246: The type or namespace name 'Logger' could not be found (are you missing a using directive or an assembly reference?) 
C:\projects\benchmarkdotnet-8ww0x\BenchmarkDotNet\BenchmarkDotNet\Loggers\MSBuildConsoleLogger.cs(16,30): error CS0115: 'MsBuildConsoleLogger.Initialize(IEventSource)': no suitable method found to override 
 
Compilation failed. 
    2 Warning(s) 
    4 Error(s) 
 
Time elapsed 00:00:01.7242660 
 
Command exited with code 1
```

--- a/.gitignore
+++ b/.gitignore
@@ -41,3 +41,4 @@ nuget.exe
 **/project.lock.json
 testsOutput/*
 .vs/restore.dg
+BDN.Auto

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,0 +1,77 @@
+#---------------------------------#
+#      general configuration      #
+#---------------------------------#
+
+# version format
+version: 0.9.7.{build}
+
+# branches to build
+branches:
+  # blacklist
+  except:
+    - gh-pages
+
+# Do not build on tags (GitHub only)
+skip_tags: true
+
+#---------------------------------#
+#    environment configuration    #
+#---------------------------------#
+
+# scripts that are called at very beginning, before repo cloning
+init:
+  - git config --global core.autocrlf input
+
+# scripts that run after cloning repository
+install:
+  # remove this when nuget 3.5 will be available on appveyor
+  - ps: Start-FileDownload "https://dist.nuget.org/win-x86-commandline/v3.5.0-beta/NuGet.exe"
+
+# enable patching of AssemblyInfo.* files
+assembly_info:
+  patch: true
+  file: AssemblyInfo.*
+  assembly_version: "{version}"
+  assembly_file_version: "{version}"
+  assembly_informational_version: "{version}"
+
+#---------------------------------#
+#       build configuration       #
+#---------------------------------#
+
+# build platform, i.e. x86, x64, Any CPU. This setting is optional.
+platform: Any CPU
+
+# build Configuration, i.e. Debug, Release, etc.
+configuration: Release
+
+os: Visual Studio 2015
+
+before_build:
+  - ps: .\build\version.ps1
+  - .\nuget.exe restore
+  - dotnet --info
+
+build:
+  parallel: true                  # enable MSBuild parallel builds
+  project: BenchmarkDotNet.sln    # path to Visual Studio solution or project
+
+after_build:
+  - dotnet pack .\BenchmarkDotNet\project.json --configuration Release
+  
+#---------------------------------#
+#       tests configuration       #
+#---------------------------------#
+before_test:
+  - dotnet test .\BenchmarkDotNet.Tests\project.json --configuration Release
+  - dotnet test .\BenchmarkDotNet.IntegrationTests\project.json --configuration Release
+
+test:
+  assemblies:
+    - BenchmarkDotNet.Tests\bin\$(configuration)\net451\win7-x64\BenchmarkDotNet.Tests.dll
+    - BenchmarkDotNet.IntegrationTests\bin\$(configuration)\net451\win7-x64\BenchmarkDotNet.IntegrationTests.dll
+    - BenchmarkDotNet.IntegrationTests.Classic\bin\$(configuration)\BenchmarkDotNet.IntegrationTests.Classic.exe
+
+artifacts:
+  - path: '**\BenchmarkDotNet.*.nupkg' # find all NuGet packages recursively
+

--- a/build/version.ps1
+++ b/build/version.ps1
@@ -1,0 +1,46 @@
+param(
+    [string]
+    $Version = $env:APPVEYOR_BUILD_VERSION
+)
+
+$projects = Get-ChildItem -Path $Path -Directory |
+                    foreach { "$($_.Fullname)\project.json" } |
+                    Where-Object { Test-Path $_ }
+
+function Set-DotnetProjectVersion
+{
+    param(
+        $project,
+        $version
+    )
+
+    $changed = $false
+    $json = Get-Content -Raw -Path $project | ConvertFrom-Json
+    if($json.version)
+    {
+        Write-Host "Setting version $version on project $project"
+        $json.version = $version
+        $changed = $true
+    }
+    
+    if($json.dependencies.BenchmarkDotNet.version)
+    {
+        $json.dependencies.BenchmarkDotNet.version = $version
+        $changed = $true
+    }
+    
+    if($changed)
+    {
+        $json | ConvertTo-Json -depth 999 | Out-File $project
+    }
+}
+
+if(-not ([string]::IsNullOrEmpty($Version)))
+{
+    Write-Host "Building version $Version"
+
+    foreach($project in $projects)
+    {
+        Set-DotnetProjectVersion $project $Version
+    }
+}

